src/akar/util.clj

```diff
@@ -0,0 +1,42 @@
+(ns akar.util
+  (:require [clojure.walk :as cw]))
+
+(defn postwalk-fn-replace
+  "Calls the matching-fn on every expression and if matching-fn
+  returned true, replaces the expression by the result of calling
+  replacing-fn with the given expr as the parameter"
+  [matching-fn replacing-fn form]
+  (cw/postwalk (fn [x]
+                   (if (matching-fn x)
+                     (replacing-fn x)
+                     x))
+                 form))
+
+
+(defmacro def-trampolined-recur
+  "For recursive functions using recur along with akar match clauses,
+  this macro generates a tail call optimized function body using
+  trampoline.
```

Comment:
As per my understanding, `trampoline` merely swaps the stack for the heap. It doesn't actually perform a tail call optimisation. Could you please verify this, and if true, correct the documentation?

Reply:
We could also avoid mentioning Akar here entirely as this utility should be useful even outside Akar context.

Reply:
Makes sense, removing Akar from the documentation.
You are right, `trampoline`  swaps the stack for the heap. It basically uses `recur` until the return value is not a function. Updating the doc.